demos/projects/ESPRESSIF/esp32/README.md

```diff
@@ -0,0 +1,284 @@
+# Connect an ESPRESSIF ESP32 using Azure IoT Middleware for FreeRTOS
+
+## What you need
```

Comment:
Need to add an option to kconfig to pick which demos/sample_azure_iot* to use (pnp and non-pnp).
Pnp must be enabled by default.

Reply:
Done.